nomad/structs/structs_test.go

```diff
@@ -8190,18 +8191,20 @@ func TestNewIdentityClaims(t *testing.T) {
 		// group: no consul.
 		// task:  no consul, no vault.
 		"job/group/task/default-identity": {
-			Namespace: "default",
-			JobID:     "job",
-			TaskName:  "task",
+			NamespacedJobID: "default::job",
+			Namespace:       "default",
+			JobID:           "job",
+			TaskName:        "task",
 			Claims: jwt.Claims{
 				Subject:  "global:default:job:group:task:default-identity",
```

Comment:
Why isn't the `sub` claim sufficient for this use case?

Reply:
Just noticed OP addressed this somewhat so I asked for some more details on the issue: https://github.com/hashicorp/nomad/issues/23510#issuecomment-2249064479